contracts/mainnet/connectors/liquity/main.sol

```diff
@@ -0,0 +1,426 @@
+pragma solidity ^0.7.6;
+
+/**
+ * @title Liquity.
+ * @dev Lending & Borrowing.
+ */
+import {
+    BorrowerOperationsLike,
+    TroveManagerLike,
+    StabilityPoolLike,
+    StakingLike,
+    CollateralSurplusLike
+} from "./interface.sol";
+import { Stores } from "../../common/stores.sol";
+import { Helpers } from "./helpers.sol";
+import { Events } from "./events.sol";
+import "hardhat/console.sol";
+
+abstract contract LiquityResolver is Events, Helpers {
+    BorrowerOperationsLike internal constant borrowerOperations =
+        BorrowerOperationsLike(0x24179CD81c9e782A4096035f7eC97fB8B783e007);
+    TroveManagerLike internal constant troveManager =
+        TroveManagerLike(0xA39739EF8b0231DbFA0DcdA07d7e29faAbCf4bb2);
+    StabilityPoolLike internal constant stabilityPool =
+        StabilityPoolLike(0x66017D22b0f8556afDd19FC67041899Eb65a21bb);
+    StakingLike internal constant staking =
+        StakingLike(0x4f9Fbb3f1E99B56e0Fe2892e623Ed36A76Fc605d);
+    CollateralSurplusLike internal constant collateralSurplus =
+        CollateralSurplusLike(0x3D32e8b97Ed5881324241Cf03b2DA5E2EBcE5521);
+    
+    // Prevents stack-too-deep error
+    struct AdjustTrove {
+        uint maxFeePercentage;
+        uint withdrawAmount;
+        uint depositAmount;
+        uint borrowAmount;
+        uint repayAmount;
+        bool isBorrow;
+    }
+
+    /* Begin: Trove */
+
+    /**
+     * @dev Deposit native ETH and borrow LUSD
+     * @notice Opens a Trove by depositing ETH and borrowing LUSD
+     * @param depositAmount The amount of ETH to deposit
+     * @param maxFeePercentage The maximum borrow fee that this transaction should permit 
+     * @param borrowAmount The amount of LUSD to borrow
+     * @param upperHint Address of the Trove near the upper bound of where the user's Trove will now sit in the ordered Trove list
```

Comment:
minor nit: this is the ideal situation, but it could happen that the list shifts between tx broadcast and mining, such that the hints are out of date and the system needs to step beyond the upper or lower hint to get the right spot.  Or if the wrong hints were passed anyway.

So I guess we should say they are estimates of bounds - or just "where the user's Trove _should_ now sit"